File: string-paradigm-api/src/main/java/com/jesperancinha/string/paradigm/api/SegmentTreeModel.java
package com.jesperancinha.string.paradigm.api;

import java.util.List;

import javax.swing.tree.DefaultMutableTreeNode;
import javax.swing.tree.TreePath;

public interface SegmentTreeModel {
	DefaultMutableTreeNode findChild(DefaultMutableTreeNode parent, String subDependency);

	DefaultMutableTreeNode addCombinationNode(DefaultMutableTreeNode parent, Dependency dependency, int codeIndex);

	void clear();
	
	List<String> getDependencies(TreePath path);
	
	DefaultMutableTreeNode addCombination(Dependency dependency);
}

File: string-paradigm-api/src/main/kotlin/org/jesperancinha/string/paradigm/api/SegmentTreeModel.kt
package org.jesperancinha.string.paradigm.api

import javax.swing.tree.DefaultMutableTreeNode
import javax.swing.tree.TreePath

interface SegmentTreeModel {
    fun findChild(parent: DefaultMutableTreeNode, subDependency: String): DefaultMutableTreeNode
    fun addCombinationNode(
        parent: DefaultMutableTreeNode,
        dependency: Dependency,
        codeIndex: Int
    ): DefaultMutableTreeNode

    fun clear()
    fun getDependencies(path: TreePath): List<String>
    fun addCombination(dependency: Dependency): DefaultMutableTreeNode
}